app/controllers/uploads_controller.rb

```diff
@@ -9,48 +9,61 @@ def index
   end
 
   def new
-    csv_type = params[:csv_type]
-
-    @upload = Upload.new(csv_type: csv_type)
-    @upload.skip_lines = defaults(csv_type)['skip_lines']
-
+    @upload = new_upload(params[:csv_type])
     return if @upload.csv_type.blank? || @upload.csv_type_check?
```

Comment:
I'm confused on this, if you return without rendering how will a user know what happened? You should at least render a flash message or something.  Are you sure you don't mean 'render new and return' ?